### PR TITLE
*nix: Kill process group when main process exits

### PR DESCRIPTION
**Pull request by mook-as**
This adds a `rdctl internal process wait-kill` command which waits for the given pid to exit, and then kills its whole process group.  This is used to kill any extension-spawned processes (as long as they haven't detached into a new process group).  This is only needed on linux / darwin because Windows has a different implementation where extension-spawned processes are moved into a job that terminates when the main process terminates instead.

This is the equivalent to #7838 but implemented completely differently.

### CODE REVIEW

File: pkg/rancher-desktop/main/extensions/manager.ts [hunk 1]
```diff
@@ -118,6 +127,21 @@ export class ExtensionManagerImpl implements ExtensionManager {
...
+      // process to exit and then kill all processes in this process group.
+      // We don't do this on Windows as we have a different (job-based)
+      // implementation there that will trigger automatically when we exit,
+      // based on `wsl-helper process spawn`.
+      const proc = spawn(
+        executable('rdctl'),
+        ['internal', 'process', 'wait-kill', `--pid=${ process.pid }`],
+        { stdio: ['ignore', await console.fdStream, await console.fdStream], detached: true });
+
+      proc.unref();
```
[the remark this answers is not in the document]
Comment (mook-as) [reply pkg/rancher-desktop/main/extensions/manager.ts:141]: We don't wait for the process to exit, so there shouldn't be any, correct.

File: src/go/rdctl/cmd/internalProcessWaitKill.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	Short: "Wait for a process and then kill of the processes in its group.",
+	Long: `rdctl internal process wait-kill waits for the given process to exit,
+then kills all processes in the same process group.`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cmd.SilenceUsage = true
+		pid, err := cmd.Flags().GetInt("pid")
+		if err != nil {
+			return fmt.Errorf("failed to get process ID: %w", err)
+		}
+		pgid, err := unix.Getpgid(pid)
```
[the remark this answers is not in the document]
Comment (mook-as) [reply src/go/rdctl/cmd/internalProcessWaitKill.go:44]: Per the [man page](https://manpages.opensuse.org/Tumbleweed/man-pages/getpgid.2.en.html):
> getpgid(), and the BSD-specific getpgrp() return a process group on success. On error, -1 is returned, and errno is set to indicate the error.

This means `err` will always be non-`nil` in the error case, so the `if err != nil` check right below will catch any errors.

File: src/go/rdctl/pkg/process/process_darwin.go [hunk 3]
```diff
@@ -83,3 +83,27 @@ func TerminateProcessInDirectory(directory string, force bool) error {
...
 }
+
+// Block and wait for the given process to exit.
+func WaitForProcess(pid int) error {
+	queue, err := unix.Kqueue()
+	if err != nil {
+		return fmt.Errorf("failed to initialize process monitoring: %w", err)
+	}
+	defer func() {
+		_ = unix.Close(queue)
```
[the remark this answers is not in the document]
Comment (mook-as) [reply src/go/rdctl/pkg/process/process_darwin.go]: Will do.

File: src/go/rdctl/cmd/internalProcessWaitKill.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Nino-K) [reply src/go/rdctl/cmd/internalProcessWaitKill.go:44]: Apologies for the confusion in marking the wrong line. What I meant to ask is whether the pid value obtained from `pid, err := cmd.Flags().GetInt("pid")` should be validated?

[the remark this answers is not in the document]
Comment (mook-as) [reply src/go/rdctl/cmd/internalProcessWaitKill.go:44]: The user input can be:
- Not a valid int: `err` is non-`nil`, and we return `failed to get process ID: %w`
- A valid pid: everything is fine
- A valid int that's not a valid pid: `unix.Getgpid` will return (`-1`, `ESRCH`), and we return `failed to get process group id for...`

**Review by Nino-K — APPROVED** [review]

**Pull request merged**